cpp/include/cudf/reduction.hpp

```diff
@@ -0,0 +1,104 @@
+  SCAN_MIN,                 ///< Computes the prefix scan of maximum operation of all values for the column
+  SCAN_MAX,                 ///< Computes the prefix scan of maximum operation of all values for the column
+  SCAN_PRODUCT,             ///< Computes the prefix scan of multiplicative product operation of all values for the column
+} scan_op;
+
+
+
+namespace reduction {
+/**
+ * @brief These enums indicate the supported reduction operations that can be
+ * performed on a column
+ */
+enum operators {
+  SUM = 0,        ///< Computes the sum of all values in the column
+  MIN,            ///< Computes the minimum of all values in the column
+  MAX,            ///< Computes the maximum of all values in the column
+  PRODUCT,        ///< Computes the multiplicative product of all values in the column
+  SUMOFSQUARES,   ///< Computes the sum of squares of the values in the column
+  MEAN,           ///< Computes the arithmetic mean of the values in the column
+  VAR,            ///< Computes the variance of the values in the column
+  STD,            ///< Computes the standard deviation of the values in the column
+  ANY,            ///< Computes to true if any of the values are non-zero/true
+  ALL,            ///< Computes to true if all of the values are non-zero/true
+};
+}  // namespace reduction
+
+
+/** --------------------------------------------------------------------------*
+ * @brief  Computes the reduction of the values in all rows of a column
+ * This function does not detect overflows in reductions.
+ * Using a higher precision `data_type` may prevent overflow.
+ * Only `min` and `max` ops are supported for reduction of non-arithmetic
+ * types (timestamp, string...).
+ * The null values are skipped for the operation.
+ * If the column is empty, the member is_valid of the output gdf_scalar
+ * will contain `false`.
+ *
+ * @param[in] col Input column view
+ * @param[in] op  The operator applied by the reduction
+ * @param[in] output_dtype  The computation and output precision.
+ *     `dtype` must be a data type that is convertible from the input dtype.
+ *     TODO(karthikeyann) update after bool support
+ *     If the input column has arithmetic type or cudf::bool8 type,
+ *     output_dtype can be any arithmetic type or cudf::bool8 type.
+ *     For `mean`, `var` and `std` ops, a floating point type must be specified.
+ *     If the input column has non-arithmetic type
+ *     (timestamp, string...), the same type must be specified.
+ * @param[in] ddof Delta Degrees of Freedom: the divisor used in calculation of `std` and `var` is `N - ddof`, where `N` is the population size.`
+ *
+ * @returns  gdf_scalar the result value
+ * If the reduction fails, the member is_valid of the output gdf_scalar
+ * will contain `false`.
+ * ----------------------------------------------------------------------------**/
+gdf_scalar reduce(const column_view& col,
+                  cudf::experimental::reduction::operators op,
```

Comment:
No need for fully qualified namespaces when you are in that namespace...

Reply:
fixed.